src/models/converters/balances.rs

```diff
@@ -2,11 +2,30 @@ use crate::models::backend::balances::Balance as BalanceDto;
 use crate::models::backend::chains::NativeCurrency;
 use crate::models::service::balances::Balance;
 use crate::providers::info::{TokenInfo, TokenType};
+use num_traits::{FromPrimitive, Pow};
 
 impl BalanceDto {
-    pub fn to_balance(&self, usd_to_fiat: f64, native_coin: &NativeCurrency) -> Balance {
-        let fiat_conversion = self.fiat_conversion.parse::<f64>().unwrap_or(0.0) * usd_to_fiat;
-        let fiat_balance = self.fiat_balance.parse::<f64>().unwrap_or(0.0) * usd_to_fiat;
+    pub fn to_balance(
+        &self,
+        token_to_usd: f64,
+        usd_to_fiat: f64,
+        native_coin: &NativeCurrency,
+    ) -> Balance {
+        let token_decimals = self
+            .token
+            .as_ref()
+            .and_then(|token| Some(token.decimals))
+            .unwrap_or(native_coin.decimals);
+
+        // We try to convert the token_decimals to f64, if it fails we fallback to the native currency decimals (f64), if it fails the function panics
+        let decimal_exp =
+            f64::from_u64(token_decimals).unwrap_or(f64::from_u64(native_coin.decimals).unwrap());
```

Comment:
Why `panic` instead? We used to have `0.0` so that we don't get rid of the entire reponse if one parsing fails.

Reply:
Token decimals as `0` might increase a lot the total balance giving the user the wrong data. I think failing here is reasonable since it relies on the native currency as a backup. If that fails then something might be really off.

Reply:
We could find a more sensible default for decimals though. Maybe `18`? I think maybe this should be discussed just to be sure that we are in sync with the expected behaviour 🤔

Reply:
I would prefer `0.0` as a default and not panicking. I think for cases that wa cannot parse something using a `f64` values couldn't be very high so it won't impact user balance. This is an exception we can recover from, I would reserve `panic` for situations in where we can not